native/cocos/renderer/gfx-vulkan/VKBuffer.cpp

```diff
@@ -73,18 +73,32 @@ void CCVKBuffer::createBufferView(uint32_t range) {
 }
 
 void CCVKBuffer::doDestroy() {
+    if (_gpuBuffer && !_isBufferView) {
+        CCVKDevice::getInstance()->gpuBarrierManager()->cancel(_gpuBuffer);
+        CCVKDevice::getInstance()->gpuBufferHub()->erase(_gpuBuffer);
+    }
+
+    if (_gpuBufferView && !hasFlag(_flags, BufferFlagBit::DISABLE_RESIZE)) {
```

Comment:
I remember buffer view cannot be resized?

Reply:
This is a whole range gpu buffer view of the `Buffer`.